library/psa_crypto.c

```diff
@@ -4724,20 +4724,22 @@ static void psa_des_set_key_parity( uint8_t *data, size_t data_size )
 */
 #if defined(MBEDTLS_PSA_BUILTIN_KEY_TYPE_ECC_KEY_PAIR) || \
     defined(MBEDTLS_PSA_BUILTIN_KEY_TYPE_ECC_PUBLIC_KEY) || \
-    defined(MBEDTLS_PSA_ACCEL_KEY_TYPE_ECC_KEY_PAIR) || \
-    defined(MBEDTLS_PSA_ACCEL_KEY_TYPE_ECC_PUBLIC_KEY)
+    defined(MBEDTLS_PSA_BUILTIN_ALG_ECDSA) || \
```

Comment:
Preexisting: the conditions on `mbedtls_ecc_group_of_psa` need to be changed as well. Since it's a public API function now, it should be defined whenever it makes sense, which is `defined(MBEDTLS_ECP_C) && defined(PSA_WANT_KEY_TYPE_ECC_PUBLIC_KEY)`. (Or `defined(PSA_WANT_ECC_KEY_TYPE_ECC_KEY_PAIR)`, but this implies that `xxx_PUBLIC_KEY` is also defined, so it doesn't need to be checked.)

Still an issue here: whether ECDSA and ECDH are supported is not relevant for ECC key derivation. The only condition here should be `MBEDTLS_PSA_BUILTIN_KEY_TYPE_ECC_KEY_PAIR`. (Not `PUBLIC_KEY` since you can't derive a public key.)